### PR TITLE
Track non-primitive args to calls returned #1172

### LINKED ISSUE #1172: Fix anonymous class wrong stack allocation with EA
>David Lloyd: in the anonymous class assigned to jdk.internal.misc.TerminatingThreadLocal#REGISTRY, the new IdentityHashMap<>(4)) is allocated on the stack

### PR DESCRIPTION
**Pull request by galderz**
Fixes #1172 

If new allocations get passed to a call that is returned, at the method level we can't guarantee that they don't escape. So, make sure those are marked for tracking as argument escape.

### CODE REVIEW

File: plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisIntraMethodBuilder.java [hunk 1]
```diff
@@ -154,14 +159,24 @@ public void startMethod(List<ParameterValue> arguments) {
     public BasicBlock return_(Value value) {
         final BasicBlock result = super.return_(value);
 
-        // Skip primitive values truncated, they are not objects
-        if (!(value instanceof Truncate)) {
+        if (value instanceof Call call && !isPrimitive(call.getType())) {
```
Comment (dmlloyd) [inline plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisIntraMethodBuilder.java:162]: Other types of calls that can return are `Invoke` and `CallNoSideEffects`; however `CallNoSideEffects` is not allowed to modify memory already so you can exclude that one. But you do need to consider `Invoke`.

Comment (galderz) [reply plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisIntraMethodBuilder.java:162]: Hmmm, is that really possible? `Invoke` is not of type `Value`.

Comment (galderz) [reply plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisIntraMethodBuilder.java:162]: Or you mean `invokeNoReturn`?

Comment (dmlloyd) [reply plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisIntraMethodBuilder.java:162]: It is; to get the return value of an `Invoke`, you have to call `getReturnValue()` on it.  The reason it is like this is that it is the only terminator which has a value.

Comment (galderz) [reply plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisIntraMethodBuilder.java:162]: I'll handle `Invoke` once https://github.com/qbicc/qbicc/pull/1233 has been fixed.

Comment (galderz) [reply plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisIntraMethodBuilder.java:162]: Sorry for delay, I'm adding `Invoke`...

**Comment by galderz** [comment]
@dmlloyd added `Invoke`

**Review by dmlloyd — APPROVED** [review]
Thanks Galder!

**Pull request merged**